sklearn/tests/test_calibration.py

```diff
@@ -618,3 +621,169 @@ def test_calibration_votingclassifier():
     calib_clf = CalibratedClassifierCV(base_estimator=vote, cv="prefit")
     # smoke test: should not raise an error
     calib_clf.fit(X, y)
+
+
+@pytest.fixture(scope="module")
+def iris_data():
+    return load_iris(return_X_y=True)
+
+
+@pytest.fixture(scope="module")
+def iris_data_binary(iris_data):
+    X, y = iris_data
+    return X[y < 2], y[y < 2]
+
+
+def test_calibration_display_validation(pyplot, iris_data, iris_data_binary):
+    X, y = iris_data
+    X_binary, y_binary = iris_data_binary
+
+    reg = LinearRegression().fit(X, y)
+    msg = "'estimator' should be a fitted classifier"
+    with pytest.raises(ValueError, match=msg):
+        CalibrationDisplay.from_estimator(reg, X, y)
+
+    clf = LinearSVC().fit(X, y)
+    msg = "response method predict_proba is not defined in"
+    with pytest.raises(ValueError, match=msg):
+        CalibrationDisplay.from_estimator(clf, X, y)
+
+    clf = LogisticRegression()
+    with pytest.raises(NotFittedError):
+        CalibrationDisplay.from_estimator(clf, X, y)
+
+
+@pytest.mark.parametrize("constructor_name", ["from_estimator", "from_predictions"])
+def test_calibration_display_non_binary(pyplot, iris_data, constructor_name):
+    X, y = iris_data
+    clf = DecisionTreeClassifier()
+    clf.fit(X, y)
+    y_prob = clf.predict_proba(X)
```

Comment:
I assume that we should only pass the probability of the positive class here.

```suggestion
    y_prob = clf.predict_proba(X)[:, 1]
```

Reply:
I think the data is not binary (iris - 4 classes?), it's checking that y should be 1-d

Reply:
iris would be 3 classes. I assume that we are using `y_true` to determine the multiclass problem then by looking at the unique labels. We always expect `y_prob` to be a 1-D array if I am not wrong.

I don't know what type of error we raise when `y_true` is binary and that we pass a 2-D ` y_prob` then.

Reply:
Ah you're right, the error is raised by `calibration_curve`: 

https://github.com/scikit-learn/scikit-learn/blob/cf286be4f3f5bb9b604efd068aaa87dc303bb4ac/sklearn/calibration.py#L908-L912

Reply:
I guess the aim of the test is to check what arises when you give the function output from multiclass classification. `y_true` is multiclass and 1d `y_prob` is not 1d, thus the error arises from:

https://github.com/scikit-learn/scikit-learn/blob/460c9610acb2163ca1e56b042b7b84b9352248f8/sklearn/calibration.py#L898